src/transactions/TransactionFrame.cpp

```diff
@@ -601,13 +641,36 @@ TransactionFrame::checkSorobanResources(
              makeU64SCVal(config.txMaxWriteBytes())});
         return false;
     }
-    if (readEntries.size() + writeEntries.size() >
-        config.txMaxDiskReadEntries())
+
+    uint32_t numDiskReads;
+    if (protocolVersionStartsFrom(ledgerVersion, AUTO_RESTORE_PROTOCOL_VERSION))
+    {
+        numDiskReads = getNumDiskReadEntries(resources, getResourcesExt());
+
+        auto totalReads = resources.footprint.readOnly.size() +
+                          resources.footprint.readWrite.size();
+        if (totalReads > config.txMaxInMemoryReadEntries())
+        {
+            diagnosticEvents.pushError(
+                SCE_STORAGE, SCEC_EXCEEDED_LIMIT,
+                "transaction total-entry-read resources exceed network config "
```

Comment:
nit: Since this has been renamed to maxFootprintEntries, the error message should stay the same ('the number of entries in transaction footprint exceeds the network config limit')